crates/proto/src/domain/errors.rs

```diff
@@ -7,16 +7,36 @@ impl SubmitProvenTransactionError {
 
     pub fn tonic_code(&self) -> tonic::Code {
         match self {
-            SubmitProvenTransactionError::InternalError
-            | SubmitProvenTransactionError::Unspecified => tonic::Code::Internal,
+            SubmitProvenTransactionError::Internal | SubmitProvenTransactionError::Unspecified => {
+                tonic::Code::Internal
+            },
             _ => tonic::Code::InvalidArgument,
         }
     }
 
     pub fn is_internal(&self) -> bool {
         matches!(
             self,
-            SubmitProvenTransactionError::InternalError | SubmitProvenTransactionError::Unspecified
+            SubmitProvenTransactionError::Internal | SubmitProvenTransactionError::Unspecified
         )
     }
 }
+
+impl SubmitProvenBatchError {
+    pub fn api_code(&self) -> u8 {
+        *self as u8
+    }
+
+    pub fn tonic_code(&self) -> tonic::Code {
+        match self {
+            SubmitProvenBatchError::Internal | SubmitProvenBatchError::Unspecified => {
+                tonic::Code::Internal
+            },
+            SubmitProvenBatchError::DeserializationFailed => tonic::Code::InvalidArgument,
+        }
```

Comment:
We could probably combine things a bit by (and the other one)

```suggestion
        if self.is_internal() {
            tonic::Code::Internal
        } else {
            tonic::Code::InvalidArgument
        }
```